storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4238,6 +4276,68 @@ bool ha_rocksdb::init_with_fields() {
   DBUG_RETURN(false); /* Ok */
 }
 
+
+/*
+  If the key is a TTL key, we may need to filter it out.
+
+  The purpose of read filtering for tables with TTL is to ensure that
+  during a transaction a key which has expired already but not removed by
+  compaction yet is not returned to the user.
+
+  Without this the user might be hit with problems such as disappearing
+  rows within a transaction, etc, because the compaction filter ignores
+  snapshots when filtering keys.
+*/
+int ha_rocksdb::should_hide_ttl_rec(const rocksdb::Slice &ttl_rec_val,
+                                    const int64_t curr_ts) const {
+  DBUG_ASSERT(m_pk_descr != nullptr);
+  DBUG_ASSERT(m_pk_descr->has_ttl());
+
+  /*
+    Curr_ts can only be 0 if there are no snapshots open.
+    should_hide_ttl_rec can only be called when there is >=1 snapshots, unless
+    we are filtering on the write path (single INSERT/UPDATE) in which case
+    we are passed in the current time as tx->m_ttl_filter_timestamp.
+  */
+  if (curr_ts == 0) {
+    sql_print_error("MyRocks: No snapshot open or timestamp passed in call to "
+                    "should_hide_ttl_rec, Aborting.");
+    abort_with_stack_traces();
```

Comment:
I created `m_ttl_filter_timestamp` keeping code for the future (secondary keys) in mind.  I think we will need it anyways.  In the event that an insert does not have a snapshot open, we still need some consistent time to filter by throughout the TX or we will get race conditions between pk and SK. If we use std::time() then it's possible that different times will be used in `should_hide_ttl_rec` calls in the write path between pk and sk.

Though, I can't immediately think of a case off the top of my head.  Will think some more about it.